frontend/src/fileReceiver.js

```diff
@@ -20,6 +20,11 @@ class FileReceiver extends EventEmitter {
         };
 
         xhr.onload = function(e) {
+
+          if (xhr.status === 404) {
+            reject(new Error('The file has expired, or has already been deleted.'));
```

Comment:
Does this need a `return` statement to prevent leaking into the code path(s) below?